lib/src/git.rs

```diff
@@ -902,8 +903,16 @@ impl<'a> RemoteCallbacks<'a> {
                         });
                     }
 
-                    if let Some(ref mut cb) = self.get_ssh_key {
-                        if let Some(path) = cb(username) {
+                    if ssh_key_paths_to_try.is_none() {
+                        if let Some(ref mut cb) = self.get_ssh_keys {
```

Comment:
nit: `.get_or_insert_with()` could be used. If `get_ssh_keys()` isn't provided, the paths can be initialized to `vec![]`.

Reply:
Thanks, that's much better.